pkgs/tools/networking/doggo/default.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  runCommand,
+  installShellFiles,
+}: let
+  buildDate = "2022-08-15T21:36:49";
+  buildCommit = "2cf9e7b";
+in
+  with lib;
+    buildGoModule rec {
+      pname = "doggo";
+      version = "0.5.4";
+      src = fetchFromGitHub {
+        owner = "mr-karan";
+        repo = pname;
+        rev = "v${version}";
+        sha256 = "sha256-6jNs8vigrwKk47Voe42J9QYMTP7KnNAtJ5vFZTUW680=";
+      };
+
+      vendorSha256 = "sha256-pyzu89HDFrMQqYJZC2vdqzOc6PiAbqhaTgYakmN0qj8=";
+      nativeBuildInputs = [installShellFiles];
+      subPackages = ["cmd/doggo"];
+
+      ldflags = [
+        "-w -s"
+        "-X main.buildVersion=${buildCommit}"
+        "-X main.buildDate=${buildDate}"
+      ];
+
+      postInstall = ''
+        installShellCompletion --cmd doggo --fish --name doggo.fish completions/doggo.fish
+        installShellCompletion --cmd doggo --zsh --name _doggo completions/doggo.zsh
+      '';
+
+      meta = {
```

```suggestion
      meta = with lib; {
```

and remove the top-level `with lib;` on line 11, it's not needed there.